ksql-engine/src/main/java/io/confluent/ksql/schema/inference/DefaultSchemaInjector.java

```diff
@@ -47,20 +61,25 @@ public DefaultSchemaInjector(final TopicSchemaSupplier schemaSupplier) {
 
   @SuppressWarnings("unchecked")
   @Override
-  public <T extends Statement> PreparedStatement<T> forStatement(
-      final PreparedStatement<T> statement
+  public <T extends Statement> ConfiguredStatement<T> inject(
+      final ConfiguredStatement<T> statement
   ) {
     if (!(statement.getStatement() instanceof AbstractStreamCreateStatement)) {
       return statement;
     }
 
-    return forCreateStatement((PreparedStatement<AbstractStreamCreateStatement>) statement)
+    final Optional<ConfiguredStatement<T>> statementOptional =
```

Comment:
I would personally find this easier to grok if all this was done inside `forCreateStatement` (and `forCreateStatement` just  returned ConfiguredStatement)